src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_controller.js

```diff
@@ -0,0 +1,64 @@
+import _ from 'lodash';
+
+const parentPipelineAggController = function ($scope) {
+
+  $scope.safeMakeLabel = function (agg) {
+    try {
+      return agg.makeLabel();
+    } catch (e) {
+      return '- agg not valid -';
+    }
+  };
+
+  $scope.$watch('responseValueAggs', updateOrderAgg);
+  $scope.$watch('agg.params.metricAgg', updateOrderAgg);
+
+  $scope.$on('$destroy', function () {
+    if ($scope.aggForm && $scope.aggForm.agg) {
+      $scope.aggForm.agg.$setValidity('bucket', true);
+    }
+  });
+
+  $scope.rejectAgg = function (agg) {
```

Comment:
Any thoughts on my comment about the function name?  If skipped over that part, I'll just reiterate, I feel like this should be something like get* or is* so it's clear this isn't rejecting an agg, but seeing if the agg should be rejected.